### PR TITLE
Remove CPT

### LINKED ISSUE #1562: Remove CPT (should be replaced by CalcFactor)
First doing #1563

### PR DESCRIPTION
**Pull request by Affie**
close #1562

### CODE REVIEW

File: src/entities/FactorOperationalMemory.jl [hunk 1]
```diff
@@ -109,52 +77,52 @@ DevNotes
...
+    _gradients::G
+    # type used for cache
+    dummyCache::CT
+
+    #Consolidation from FMD
+    fullvariables::Vector{DFGVariable}
+
+    #Consolidation from CPT
+    # the actual particle being solved at this moment
+    particleidx::Int
```
Comment (dehann) [inline src/entities/FactorOperationalMemory.jl:121]: these Old CPT fields are specific to each particle (thread) in the future so can therefore not stay in CCW.  They should only be created in CalcFactor.

File: src/services/NumericalCalculations.jl [hunk 2]
```diff
@@ -197,10 +197,10 @@ end
...
 
 Notes  
-- Unless passed in as separate arguments, this assumes already valid in `cpt_`:
-  - `cpt_.p`
-  - `cpt_.activehypo`
-  - `cpt_.factormetadata`
+- Unless passed in as separate arguments, this assumes already valid in `ccwl`:
+  - `ccwl.p`
+  - `ccwl.activehypo`
+  - `ccwl.factormetadata`
```
Comment (dehann) [inline src/services/NumericalCalculations.jl:203]: can remove this line

**Review by dehann — APPROVED** [review]
Good for removing CPT.  This PR does create the technical debt of using future thread specific memory in CCW.  So the old CPT fields will have to be removed from CCW again.

File: src/entities/FactorOperationalMemory.jl (hunk 1 above)
Comment (Affie) [reply src/entities/FactorOperationalMemory.jl:121]: We can discuss it at some stage. The mission here was to remove CPT.

File: src/services/NumericalCalculations.jl (hunk 2 above)
Comment (Affie) [reply src/services/NumericalCalculations.jl]: done

**Comment by Affie** [comment]
> Good for removing CPT. This PR does create the technical debt of using future thread specific memory in CCW. So the old CPT fields will have to be removed from CCW again.

Technical debt can be captured here #1616

File: src/entities/FactorOperationalMemory.jl (hunk 1 above)
Comment (dehann) [reply src/entities/FactorOperationalMemory.jl:121]: sounds good

**Comment by dehann** [comment]
thanks Johan, you can merge this one too when you are ready.

**Pull request merged**